doc.js

```diff
@@ -49,12 +49,28 @@ define('doc', ['event'], function(event) {
 		return el.className.match(new RegExp('(^|\\s)' + clazz + '($|\\s)'));
 	};
 
+	var convertNodeListToArray = function(nodeList) {
+		var length = nodeList.length;
+		var list = [];
+		for (var i = 0; i < length ; i++) {
+			var element = nodeList[i];
+			if (element.nodeType === Node.ELEMENT_NODE) {
+				list.push(element);
+			}
+		}
+		return list;
+	}
+
 	var query = function(elements) {
 		var selectedElements = [];
 
-		if (elements){
+		if (elements) {
 			if (elements instanceof Array) {
 				selectedElements = elements;
+			} else if (elements instanceof HTMLCollection) {
+				selectedElements = convertHtmlCollectionToArray(elements);
+			} else if (elements instanceof NodeList) {
+				selectedElements = convertNodeListToArray(elements);
```

Comment:
Why don't you use `Array.prototype.slice.call(elements)` for both cases? The method is unnecessary.

@tcelestino @mottam